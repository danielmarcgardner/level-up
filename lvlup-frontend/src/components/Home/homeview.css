.home-header {
  text-align: center;
  margin-top: 0px;
  padding-top: 6rem;
  padding-bottom: 6rem;
  width: 100%;
  height: 324px;
  font-size: 9rem;
  /*color: #4A4A4A;*/
  color: #FFFFFF;
  font-weight: bold;
  font-family: 'Nova Square', sans-serif;
  /*background-repeat: no-repeat;*/
  /*background-size: cover;*/
  background-image: url("../../images/tags.jpeg");
}

.ui.small.nav.menu {
  background-color: #4A4A4A !important;
  margin-bottom: 0px;
  border-radius: 0% !important;
  border-bottom: 1px solid;
  border-top: 1px solid;
  border-color: #F2711C;
}

.hamburger {
  color: #F2711C;
  font-weight: 100 !important;
  font-size: 18px;
}

.description {
  margin-top: 50px;
  width: 80%;
  margin-bottom: 80px;
  text-align: center;
}

.demo-modal {
  margin-top: 50px;
  width: 50%;
  margin-left: 0%;
  margin-right: 10%;
  text-align: center;
  padding-right: 2em;
}

.footer{
  text-align: center;
  background-color: #4A4A4A !important;
  position: fixed;
  bottom: 0;
  width: 100%;
  height: 46px;
  color: white;
  padding: 5px 12px;
  border-radius: 0% !important;
  border-bottom: 1px solid;
  border-top: 1px solid;
  border-color: #F2711C;
}

.modalRecruiter {
  text-align: center;
  margin-bottom: 20px;
  margin-top: 20px;
}

.modalGreeting {
  margin-bottom: 10px;
}

.p-topmargin {
  margin-top: 10px;
}

.p-left {
  width: 60%;
  margin-left: 30%;
  margin-right: 10%;
  text-align: left;
  margin-top: 25px;
  margin-bottom: 25px;
}

.ui.form.email {
  margin-top: 30px !important;
}

#nova-font {
 font-family: 'Nova Square', sans-serif;
}

#intro-text {
  font-size: 24px;
  line-height: 38px;
  margin-top: 60px;
  margin-right: 10%;
  margin-left: 10%;
  margin-bottom: 60px;
  text-align: center;
  font-family: 'Nova Square', sans-serif;
}

#demo-instructions {
  margin-bottom: 20px;
}

#challenge-solution {
  padding: 60px;
  font-size: 16px;
  line-height: 32px;
  background-color: #CCCCCC;
  text-align: center;
  font-family: 'Nova Square', sans-serif;
}

#challenge-column {
  padding-right: 40px;
}

#solution-column {
  padding-left: 40px;
}

#challenge-segment {
  padding: 40px;
}

#solution-segment {
  padding: 40px;
}

.technologyCards {
  font-family: 'Nova Square', sans-serif;
  text-align: center !important;
}

.cardDiv {
  /*width: 90%;*/
  margin-left: 10%;
  margin-right: 10%
}

div.description {
  text-align: center !important;
  width: 80%;
  margin-left: 10%;
}
