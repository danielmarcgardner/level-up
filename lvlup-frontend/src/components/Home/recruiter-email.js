import React, { Component } from 'react';
import { Form } from 'semantic-ui-react';
import { Field, reduxForm } from 'redux-form';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { renderField } from '../Admin/admin-common/render-fields';
import { required, email } from '../Admin/admin-common/validations';
import { recruiter } from '../../actions/admin-signup';
import renderIf from 'render-if';

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ recruiter }, dispatch);
}

function mapStateToProps(state, ownProps) {
  return {
    recruiter: state.recruiter,
  };
}

class RecruiterEmail extends Component {
  render() {
    const { handleSubmit } = this.props;
    return (
      <div className="recruiterEmail" >
        {renderIf(!this.recruiter)(<Form className="email" onSubmit={handleSubmit(this.props.recruiter)}>
          <Form.Field inline>
            <Field
              name="email"
              component={renderField}
              type="email"
              label=" Your E-mail:"
              placeholder="Student login credentials for demo will be sent here"
              validate={[required, email]}
            />
          </Form.Field>
          <Form.Button basic color="orange">Request Login Info</Form.Button>
        </Form>,
      )}
        {renderIf(this.recruiter)(
          <p>Please check your email for GitHub Login Info for the student OAuth</p>,
      )}
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(reduxForm({
  form: 'recruiter',
})(RecruiterEmail));
