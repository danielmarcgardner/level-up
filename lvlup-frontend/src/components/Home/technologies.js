import React from 'react';
import { Grid, Card, Image } from 'semantic-ui-react';

const Technologies = () => (
  <div>
    <Grid.Column width={8} id="technologies-columnL">
      <u>The Front End Technologies Used:</u> <div>
        <Card.Group className="technologyCards">
          <Card>
            <Card.Content>
              <Image size="mini" src={require('../../images/javascript.png')} />
              <Card.Header>
                JavaScript
              </Card.Header>
              <Card.Meta>
                Front End and Back End
              </Card.Meta>
              <Card.Description>
                lvl^ is written in both the front end and back end with JavaScript.
              </Card.Description>
            </Card.Content>
          </Card>
          <Card>
            <Card.Content>
              <Image size="mini" src={require('../../images/github.png')} />
              <Card.Header>
                GitHub
              </Card.Header>
              <Card.Meta>
                Front End and Back End
              </Card.Meta>
              <Card.Description>
                GitHub was used to for collaboration on lvl^ as well as OAuth on the backend
              </Card.Description>
            </Card.Content>
          </Card>
          <Card>
            <Card.Content>
              <Image size="mini" src={require('../../images/react.png')} />
              <Card.Header>
                React
              </Card.Header>
              <Card.Meta>
                Front End
              </Card.Meta>
              <Card.Description>
                lvl^'s front end is built with React
              </Card.Description>
            </Card.Content>
          </Card>
        </Card.Group>

        {/* <Image src={require('../../images/github.png')} className="technologyImage" size="tiny" />
          <Image src={require('../../images/react.png')} className="technologyImage" size="tiny" />
          <Image src={require('../../images/react-router.png')} className="technologyImage" size="tiny" />
          <Image src={require('../../images/redux.png')} className="technologyImage" size="tiny" />
          <Image src={require('../../images/redux-forms.png')} className="technologyImage" size="tiny" />
          <Image src={require('../../images/semantic-ui.png')} className="technologyImage" size="tiny" />
          <Image src={require('../../images/moment.png')} className="technologyImage" size="tiny" />
          <Image src={require('../../images/jest.png')} className="technologyImage" size="tiny" /> */}
      </div>
    </Grid.Column>
    <Grid.Column textAlign="center" width={8} id="technologies-columnR">
      <u>The Back End Technologies Used:</u>
      {/* <div>
        <Image src={require('../../images/nodejs-icon.svg')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/express.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/knex.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/bookshelf.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/postgresql.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/jwt.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/oauth.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/passport.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/nodemailer.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/moment.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/bluebird.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/chai.png')} className="technologyImage" size="tiny" />
        <Image src={require('../../images/mocha.svg')} className="technologyImage" size="tiny" />
      </div> */}
    </Grid.Column>
  </div>
  );
export default Technologies;
