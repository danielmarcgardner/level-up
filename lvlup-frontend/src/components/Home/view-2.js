import React, { Component } from 'react';

export default class home2 extends Component {
  render() {
    return (<div>Hello</div>);
  }
}
