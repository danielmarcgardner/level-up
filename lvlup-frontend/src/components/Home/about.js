import React from 'react';
import { Grid } from 'semantic-ui-react';
import './homeview.css';

const About = props => (
  <div>
    <Grid>
      <Grid.Row />
    </Grid>
  </div>
);

export default About;
