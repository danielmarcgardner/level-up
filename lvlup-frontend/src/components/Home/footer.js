import React from 'react';
import './homeview.css';

const Footer = () => (
  <div className="footer" textAlign="center">
    Contact Us &emsp;&emsp;&emsp; || &emsp;&emsp;Enter &emsp;&emsp;|| &emsp;&emsp;&emsp;To the Top &emsp;&emsp;
  </div>
);

export default Footer;
