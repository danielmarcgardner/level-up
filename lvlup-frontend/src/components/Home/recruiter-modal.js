import React from 'react';
import { Button, Header, Icon, Modal, Image } from 'semantic-ui-react';
import RecruiterEmail from './recruiter-email';
import './homeview.css';

const RecruiterModal = () => (
  <Modal trigger={<Button color="orange" inverted>Recruiter's Click Here</Button>}>
    <Modal.Header>Recruiter Instructions</Modal.Header>
    <Modal.Content image>
      <Image wrapped size="large" src={require('../../images/logo.png')} />
      <Modal.Description>
        <Header>Welcome Recruiters! Thank you for visiting!</Header>
        <div className="modalRecruiter">
          <p>Welcome to lvl^,  the Galvanize Capstone for Daniel Gardner <a href={'https://github.com/danielmarcgardner'}><Icon name="github" /></a><a href={'https://www.linkedin.com/in/danielmarcgardner/'}><Icon name="linkedin" /></a>,
            Thomas Stang <a href={'https://github.com/tkstang'}><Icon name="github" /></a><a href={'https://www.linkedin.com/in/thomas-stang/'}><Icon name="linkedin" /></a>, and Sanjeet Uppal <a href={'https://github.com/uppal101'}><Icon name="github" /></a><a href={'https://www.linkedin.com/in/sanjeet-uppal/'}><Icon name="linkedin" /></a>.
            This site will continue to be updated with new features so please continue to checkback.</p>

          <p>If you have a GitHub Account you can sign up as a student. If you do not have a GitHub account please fillout the email field below and we will send you an email with the credentials to log into our demo account.
            For demo purposes all emails are allowed to create admin accounts. To access all demo data please select cohort g42 for cohorts. Please note that you will need to confirm your admin account via an email that will be sent to your email address.
          </p>
          <p>As a student (with your own or our demo account) you can:</p>
          <ul>
            <li>View current submissions</li>
            <li>See achievements</li>
            <li>See the rewards earned</li>
            <li>Make challenge submissions</li>
            <li>Make reward requests</li>
          </ul>
          <p>As an admin you can:</p>
          <ul>
            <li>View all pending challenge submissions</li>
            <li>View all pending reward requests</li>
            <li>Approve or deny challenge submissions and reward requests</li>
            <li>Add or Edit Challenges </li>
            <li>Add or Edit Rewards</li>
          </ul>
          <p> Please note that we will be rolling back and re-seeding our database every so often to reset and re-seed.</p>
          <p>To close this modal please hit the escape key or click outside the modal.</p>
          <div className="recruiterEmail">
            <RecruiterEmail />
          </div>
        </div>
      </Modal.Description>
    </Modal.Content>
  </Modal>
);

export default RecruiterModal;
