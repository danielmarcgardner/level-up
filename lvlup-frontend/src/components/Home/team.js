import React, { Component } from 'react';
import { Grid, Card, Image, Reveal } from 'semantic-ui-react';
import './homeview.css';

const Team = props => (
  <div className="team-cards">
    <Grid>
      <Grid.Row columns={3}>
        <Grid.Column>
          <Card className="team-card" centered raised="true" color="orange">
            <Reveal animated="move up">
              <Reveal.Content visible>
                <Image className="team-image" src="https://avatars.githubusercontent.com/tkstang" />
              </Reveal.Content>
              <Reveal.Content hidden>
                <div>Some text</div>
              </Reveal.Content>
            </Reveal>
            <Card.Content>
              <Card.Header>
                Thomas Stang
              </Card.Header>
              <Card.Meta>
              Blah
            </Card.Meta>
              <Card.Description>
              Thomas is the man
            </Card.Description>
            </Card.Content>
          </Card>
        </Grid.Column>
        <Grid.Column>
          <Card className="team-card" centered raised="true" color="orange">
            <Reveal animated="move up">
              <Reveal.Content visible>
                <Image className="team-image" src="https://avatars.githubusercontent.com/uppal101" />
              </Reveal.Content>
              <Reveal.Content hidden>
                <div>Some text</div>
              </Reveal.Content>
            </Reveal>
            <Card.Content>
              <Card.Header>
                Sanjeet Uppal
              </Card.Header>
              <Card.Meta>
              Blah
            </Card.Meta>
              <Card.Description>
              Thomas is the man
            </Card.Description>
            </Card.Content>
          </Card>
        </Grid.Column>
        <Grid.Column>
          <Card className="team-card" centered raised="true" color="orange">
            <Reveal animated="move up">
              <Reveal.Content visible>
                <Image className="team-image" src="https://avatars2.githubusercontent.com/u/22782154?v=3" />
              </Reveal.Content>
              <Reveal.Content hidden>
                <div>Some text</div>
              </Reveal.Content>
            </Reveal>
            <Card.Content>
              <Card.Header>
                Daniel Gardner
              </Card.Header>
              <Card.Meta>
              Blah
            </Card.Meta>
              <Card.Description>
              Thomas is the man
            </Card.Description>
            </Card.Content>
          </Card>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </div>
);

export default Team;
