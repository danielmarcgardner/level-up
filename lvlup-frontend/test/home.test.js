import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from '../src/components/Home/home';

test('Home should render', () => {
  const home = shallow(
    <Home />,
  );
  expect(toJson(home)).toMatchSnapshot();
});
