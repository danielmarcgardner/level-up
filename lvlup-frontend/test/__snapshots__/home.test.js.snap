exports[`test Home should render 1`] = `
<div>
  <h1
    className="home-header">
    lvl^
  </h1>
  <div
    className="description">
    <p>
      Welcome to lvl^ the gamification of tasks that help you in your quest to become a better developer and your career. Students are given challenges and get a certain amount of points for completing the challenges. They are then able to redeem their points for rewards. The categories in both rewards and challenges should enrich the student\'s education, career search, and personal life. Students can sign in with GitHub OAuth in the upper right hand corner. Admins can sign up or sign in in the bottom right corner.
    </p>
  </div>
  <Footer />
</div>
`;
